engine/src/execution_graph/logic_controllers/BatchOrderByProcessing.h

```diff
@@ -369,19 +353,23 @@ class PartitionKernel : public kernel {
 							continue;
 						}
 
-						metadata.add_value(ral::cache::WORKER_IDS_METADATA_LABEL, dest_node.id());
-						metadata.add_value(ral::cache::CACHE_ID_METADATA_LABEL, "output_" + std::to_string(part_ids[i]) );
-						
-						node_count[dest_node.id()]++;
-						output_cache->addCacheData(std::unique_ptr<ral::cache::GPUCacheData>(new ral::cache::GPUCacheDataMetaData(table_view.clone(), metadata)),"",true);
+						send_message(std::move(table_view.clone()),
+							"true", //specific_cache
+							"output_" + std::to_string(part_ids[i]), //cache_id
+							dest_node.id(), //target_id
+							"", //total_rows
+							"", //message_id_prefix
+							true); //always_add
+
+						increment_node_count(dest_node.id(), 0);
```

Comment:
`send_message` should be the one that does `increment_node_count`

Reply:
Done. Now the increment is performed inside the function send_message.